kratos/utilities/parallel_utilities.cpp

```diff
@@ -104,31 +104,17 @@ int ParallelUtilities::InitializeNumberOfThreads()
 #endif
 }
 
-ParallelUtilities& ParallelUtilities::GetInstance()
+int& ParallelUtilities::GetNumberOfThreads()
 {
-    // Using double-checked locking to ensure thread safety in the first creation of the singleton.
-    if (!mpInstance) {
-        #ifdef KRATOS_SMP_OPENMP
-        #pragma omp critical
-        if (!mpInstance) {
-        #endif
-            Create();
-            mpInstance->msNumThreads = InitializeNumberOfThreads();
-        #ifdef KRATOS_SMP_OPENMP
-        }
-        #endif
+    if (!mspNumThreads) {
+        static int num_threads;
+        num_threads = InitializeNumberOfThreads();
+        mspNumThreads = &num_threads;
```

Comment:
```suggestion
        static int number_of_threads = InitializeNumberOfThreads();
        return number_of_threads;
```

Reply:
If you are not using the double guard, having the static pointer is not needed

Reply:
Please note that in your implementation you were initializing in all calls. In the suggested one it is done once.

Reply:
I think it was the other way round
probably easier to speak